Editor/MobiledgeXEditorWindow.cs

```diff
@@ -307,18 +322,69 @@ private async void SetupWindow()
+
+            settings.edgeEventsConfig.locationConfig.updatePattern =
+                (UpdatePattern)EditorGUILayout.EnumPopup(
+                    new GUIContent("Update Pattern", "UpdatePattern for sending client events " +
+                "\nOnStart will send one update when startEdgeEvents is called " +
+                "\nOnInterval will send updates at the specified interval " +
+                "\nOnTrigger the application is responsible for sending the events ")
+              , settings.edgeEventsConfig.locationConfig.updatePattern);
+
+            if (settings.edgeEventsConfig.locationConfig.updatePattern == UpdatePattern.OnInterval)
+            {
+                settings.edgeEventsConfig.locationConfig.updateIntervalSeconds =
+                    EditorGUILayout.IntField
+                    (new GUIContent("Update Interval(s)", "Update interval in seconds")
+                    , settings.edgeEventsConfig.locationConfig.updateIntervalSeconds);
+                settings.edgeEventsConfig.locationConfig.maxNumberOfUpdates =
+                    EditorGUILayout.IntField(
+                        new GUIContent("Max. Number of Updates",
+                    "Maximum number of updates through out the App lifetime." +
+                    " Set to -1 for updates to run till the EdgeEvents connection is closed")
+                    , settings.edgeEventsConfig.locationConfig.maxNumberOfUpdates);
+            }
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            EditorGUILayout.LabelField(new GUIContent("Latency Events Settings"), titleStyle);
+
+            settings.edgeEventsConfig.latencyConfig.updatePattern =
+                (UpdatePattern)EditorGUILayout.EnumPopup(
+                    new GUIContent("Update Pattern","UpdatePattern for sending client events " +
+                "\nOnStart will send one update when startEdgeEvents is called " +
+                "\nOnInterval will send updates at the specified interval " +
+                "\nOnTrigger the application is responsible for sending the events ")
+              , settings.edgeEventsConfig.latencyConfig.updatePattern);
+            if (settings.edgeEventsConfig.latencyConfig.updatePattern == UpdatePattern.OnInterval)
+            {
+                settings.edgeEventsConfig.latencyConfig.updateIntervalSeconds =
+                    EditorGUILayout.IntField(
+                        new GUIContent("Update Interval (s)", "Update interval in seconds")
+                        , settings.edgeEventsConfig.latencyConfig.updateIntervalSeconds);
+                settings.edgeEventsConfig.latencyConfig.maxNumberOfUpdates =
+                  EditorGUILayout.IntField(
+                      new GUIContent("Max. Number of Updates",
+                  "Maximum number of updates through out the App lifetime." +
+                  " Set to -1 for updates to run till the EdgeEvents connection is closed")
```

Comment:
In Swift, I have it set so that maxNumberOfUpdates <= 0 will update infinitely. 

It looks like your default maxNumberOfUpdates is 0. Based on this comment it seems like you won't update at all if maxNumberOfUpdates is 0, which doesn't really make sense, if your UpdatePattern is OnInterval.

Reply:
C# doesn't work like that yet. It sees zero, and says OK, which does make sense to me, as if someone wants that config, who are we to argue. In any case, if the goal is make something happen, regardless of this edge case interpretation, then 0 == infinity isn't bad behavior, just that for custom tasks, this will be different with respect to the actual default implementation unless documented. They are ultimately implementing it with Raw Events so the default is less important.